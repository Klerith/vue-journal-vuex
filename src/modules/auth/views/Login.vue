<template>
    <span class="login100-form-title p-b-41">
        Ingresar
    </span>
    <form class="login100-form validate-form p-b-33 p-t-5"
        @submit.prevent="onSubmit">

        <div class="wrap-input100 validate-input" data-validate = "Enter username">
            <input v-model="userForm.email" class="input100" type="text" placeholder="Correo" required>
            <span class="focus-input100" data-placeholder="&#xe82a;"></span>
        </div>

        <div class="wrap-input100 validate-input" data-validate="Enter password">
            <input v-model="userForm.password" class="input100" type="password" placeholder="Contraseña" required>
            <span class="focus-input100" data-placeholder="&#xe80f;"></span>
        </div>

        <div class="container-login100-form-btn m-t-32">
            <button class="login100-form-btn">
                Login
            </button>

        </div>

        <div class="container-login100-form-btn m-t-32">
            <router-link :to="{ name: 'register' }">¿No tienes cuenta?</router-link>
        </div>
    </form>
</template>

<script>
import { ref } from 'vue'
import { useRouter } from 'vue-router'
import useAuth from '../composables/useAuth'

import Swal from 'sweetalert2'

export default {
setup() {

    const router = useRouter()
    const { loginUser } = useAuth()

    const userForm = ref({
        email: 'fernando@gmail.com',
        password: '123456',
    })


    return {
            userForm,
            onSubmit: async() => {
                const { ok, message } = await loginUser( userForm.value )

                if ( !ok ) Swal.fire('Error', message, 'error' )
                else router.push({ name: 'no-entry' })
            }
        }
    }
}
</script>
