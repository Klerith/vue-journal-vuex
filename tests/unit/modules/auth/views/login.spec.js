import { shallowMount } from '@vue/test-utils'
import Login from '@/modules/auth/views/Login.vue'

import createVuexStore from '../../../mock-data/mock-store'

import Swal from 'sweetalert2'

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))



describe('Pruebas en el Login Component', () => {
    
    const store = createVuexStore({
        status: 'not-authenticated', // 'authenticated','not-authenticated', 'authenticating'
        user: null,
        idToken: null,
        refreshToken: null
    })

    store.dispatch = jest.fn()

    beforeEach(() => jest.clearAllMocks() )


    test('debe de hacer match con el snapshot', () => {
        
        const wrapper = shallowMount( Login, {
            global: {
                plugins: [ store ]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    test('credenciales incorrectas, disparar el SWAL', async() => {
        
        store.dispatch.mockReturnValueOnce({ ok: false, message: 'Error en credenciales' })

        const wrapper = shallowMount( Login, {
            global: {
                plugins: [ store ]
            }
        })

        await wrapper.find('form').trigger('submit')
        expect( store.dispatch ).toHaveBeenCalledWith('auth/signInUser', { email: '', password: ''})
        expect( Swal.fire ).toHaveBeenCalledWith('Error', 'Error en credenciales', 'error')
    })
    
    
    test('debe de redirigir a la ruta no-entry', async() => {
        
        store.dispatch.mockReturnValueOnce({ ok: true })

        const wrapper = shallowMount( Login, {
            global: {
                plugins: [ store ]
            }
        })

        const [ txtEmail, txtPassword ] = wrapper.findAll('input')
        await txtEmail.setValue('fernando@gmail.com')
        await txtPassword.setValue('123456')

        await wrapper.find('form').trigger('submit')

        expect( store.dispatch ).toHaveBeenCalledWith('auth/signInUser', { email: 'fernando@gmail.com', password: '123456' })
        expect( wrapper.router.push ).toHaveBeenCalledWith({ name: 'no-entry' })

        // console.log(wrapper.vm.userForm)
    })
    


})




