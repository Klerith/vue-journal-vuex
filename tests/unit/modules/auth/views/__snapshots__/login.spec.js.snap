// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Login Component debe de hacer match con el snapshot 1`] = `
<span class="login100-form-title p-b-41"> Ingresar </span>
<form class="login100-form validate-form p-b-33 p-t-5">
  <div class="wrap-input100 validate-input" data-validate="Enter username"><input class="input100" type="text" placeholder="Correo" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Enter password"><input class="input100" type="password" placeholder="Contraseña" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="container-login100-form-btn m-t-32"><button class="login100-form-btn"> Login </button></div>
  <div class="container-login100-form-btn m-t-32">
    <router-link-stub to="[object Object]" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </div>
</form>
`;
